src/test/java/com/epam/tc/hw2/TestExercise1HomePage.java

```diff
@@ -0,0 +1,99 @@
+        webDriver.findElement(By.id("password")).sendKeys("Jdi1234");
+        webDriver.findElement(By.id("login-button")).click();
+        // 4. Assert Username is loggined
+        WebElement username = webDriver.findElement(By.id("user-name"));
+        softAssertions.assertThat(username.isDisplayed()).isTrue();
+        softAssertions.assertThat(username.getText()).isEqualTo(expectedUserName);
+        // 5. Assert that there are 4 items on the header section are displayed and they have proper texts
+        List<WebElement> navigationBarItems = webDriver.findElements(
+            By.cssSelector("ul[class=\"uui-navigation nav navbar-nav m-l8\"]>li"));
+        softAssertions.assertThat(navigationBarItems.size()).isEqualTo(expectedNavigatorBarItems.size());
+        assertThatAllElementsExpectedlyDisplayed(navigationBarItems, expectedNavigatorBarItems);
+        // 6. Assert that there are 4 images on the Index Page and they are displayed
+        List<WebElement> benefitIcons = webDriver.findElements(By.cssSelector("span[class^=\"icons-benefit\"]"));
+        softAssertions.assertThat(benefitIcons.size()).isEqualTo(expectedIconsNumber);
+        // 7. Assert that there are 4 texts on the Index Page under icons and they have proper text
+        List<WebElement> benefitTexts = webDriver.findElements(By.cssSelector("span[class=\"benefit-txt\"]"));
+        assertThatAllElementsExpectedlyDisplayed(benefitTexts, expectedBenefitTexts);
+        // 8. Assert that there is the iframe with “Frame Button” exist
+        WebElement buttonFrame = webDriver.findElement(By.id("frame"));
+        softAssertions.assertThat(buttonFrame.isDisplayed()).isTrue();
+        // 9. Switch to the iframe and check that there is “Frame Button” in the iframe
+        webDriver.switchTo().frame(buttonFrame);
+        WebElement inFrameButton = webDriver.findElement(By.id("frame-button"));
+        softAssertions.assertThat(inFrameButton.isDisplayed()).isTrue();
+        // 10. Switch to original window back
+        webDriver.switchTo().parentFrame();
+        // 11. Assert that there are 5 items in the Left Section are displayed and they have proper text
+        List<WebElement> sidebarElements = webDriver.findElements(By.cssSelector("ul[class=\"sidebar-menu left\"]>li"));
+        softAssertions.assertThat(sidebarElements.size()).isEqualTo(expectedSidebarElements.size());
+        assertThatAllElementsExpectedlyDisplayed(sidebarElements, expectedSidebarElements);
+        // 12. Close Browser
+        webDriver.quit();
+    }
+
+    private void assertThatAllElementsExpectedlyDisplayed(List<WebElement> actualElements,
+                                                          List<String> expectedElements) {
+        for (int i = 0; i < actualElements.size(); i++) {
+            softAssertions.assertThat(actualElements.get(i).isDisplayed()).isTrue();
+            softAssertions.assertThat(actualElements.get(i).getText()).isEqualTo(expectedElements.get(i));
```

Comment:
selenium has some default method that allows us to compare collections as the collections not element by element

Reply:
Do you mean assertj? I thought about it, but for comparing I must mock List<WebElements> isn't it? I can create List<Strings> from List<WebElements> and compare two List<Strings>, but I still need to use cycle or stream().

Reply:
Fair point